sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivityViewModel.kt

```diff
@@ -58,54 +60,62 @@ class SyncActivityViewModel @Inject constructor(
     private val qrEncoder: QREncoder,
     private val recoveryCodePDF: RecoveryCodePDF,
     private val syncRepository: SyncRepository,
+    private val syncStateMonitor: SyncStateMonitor,
     private val dispatchers: DispatcherProvider,
 ) : ViewModel() {
 
     private val command = Channel<Command>(1, DROP_OLDEST)
     private val viewState = MutableStateFlow(ViewState())
+    fun commands(): Flow<Command> = command.receiveAsFlow()
 
-    fun viewState(): Flow<ViewState> = viewState.onStart {
-        viewState.emit(initViewStateThisDeviceState())
-        viewModelScope.launch(dispatchers.io()) {
-            viewState.emit(viewState.value.showDeviceListItemLoading())
-            fetchRemoteDevices()
-        }
-        observerSignedInState()
-    }.flowOn(dispatchers.io())
-
-    private fun observerSignedInState() {
-        syncRepository.isSignedInFlow().onEach { signedIn ->
-            when (signedIn) {
-                true -> {
-                    if (!viewState.value.isSignedInState()) {
-                        viewState.emit(initViewStateThisDeviceState())
-                    }
-                }
+    fun viewState(): StateFlow<ViewState> {
+        return viewState
+    }
 
-                false -> viewState.emit(signedOutState())
+    fun observeState() {
```

Comment:
This seems something we want to start doing when someone is observing `viewState()`, right?